추억저장소

theBettor opened the pull request:
## 🤷‍♂️ PR 내용(Issue 꼭 달기!)
추억저장소를 위해 2가지의 모듈을 만들었다. memorystorage와 memorystorage-detail
memorystorage에서는 간단하게 어떤 추억카드들이 있는지를 보여주고,
memorystorage-detail에서는 카드 중 선택한 한개의 memoryCardId에 대한 정보를 보여줘야한다.
그래서 특이하게 Route 데이터 클래스에서는 String타입인 memoryCardId를 변수를 갖는다.
근데 지금 memoryCardId를 잘 사용하고 있는지 모르겠다. 일단은 previewActivity를 사용하고 있기 때문에 정확한 확인이 어려운 상황이다.

-> 아직까지 에러는 없는데.. memoryCardId를 함수 파라미터(?)로 넘기는 부분을 임의로 "memoryCardId"로 지정했다.

resolved: #114

theBettor committed:
feat: 추억저장소 모듈 2개 생성

theBettor committed:
saved UNDO information for reset changes and discarded files

theBettor committed:
refactor: 추억저장소 화면 부분 마무리 #116

sealed class를 사용하여 MemoryStorageUiState를 상태별로 관리
Success, Loading, Error 상태를 명확하게 분리

theBettor committed:
feat: 추억저장소 메인앱 연결 #116

theBettor committed:
refactor: MemoryStorageDetail 부분 작업 완료

theBettor committed:
refactor : 화면 부분 마무리와 화면 연결

디자인상 플로우 설명이 명확하지 부분이 있어 일단 부분 마무리합니다.

theBettor committed:
refactor: MemoryCard 객체 사용

theBettor committed:
refactor: 2가지의 화면상태

theBettor committed:
Merge remote-tracking branch 'origin/develop' into 0423-meory_storage

theBettor committed:
temp

theBettor committed:
Merge remote-tracking branch 'origin/0423-meory_storage' into feat/memory_storage

theBettor committed:
temp

theBettor committed:
temp

theBettor committed:
Merge remote-tracking branch 'origin/backup-stash' into feat/memory_storage

theBettor committed:
fix: gradle.properties

DJDrama commented on app/src/main/java/com/teampatch/harmony/MainApp.kt:
```diff
@@ -56,6 +58,12 @@ fun MainApp(
                 }
 
+                MemoryStorageRoute::class.qualifiedName -> {
+                    NavigationItem.STORE.also {
```
also는 sideEffect에 쓰는 용도로 사용합니다.(코딩 컨벤션)

차라리 apply를 사용하는 것이 좋을 것 같습니다.

예를 들어 also는
Toast를 띄운다거나, startActivity와 같이 SideEffect(다른 놈의 상태를 바꾸는 용도로)

DJDrama commented on app/src/main/java/com/teampatch/harmony/MainNavHost.kt:
```diff
@@ -193,5 +195,15 @@ fun MainNavHost(
         )
+
+        addMemoryStorageDetailScreen(
+            onBackRequest = { navController.popBackStack() },
```
여기선 왜 popBackStack을 사용했을까요?? 아래는 navigateUp을 사용했구

DJDrama commented on core/designsystem/src/main/java/com/teampatch/core/designsystem/component/MemoryInfoView.kt:
```diff
@@ -0,0 +1,149 @@
+                        fontSize = 12.sp,
+                        maxLines = 1
+                    )
+                }
```
```
@Composable
fun CircleText(
  modifier: Modifier = Modifier,
  text: String
){
    // 첫 번째 원형 텍스트
                Box(
                    modifier = Modifier
                        .size(40.dp) // 원 크기
                        .background(MainGreen, shape = CircleShape) // 원 배경색과 둥근 모양
                        .padding(1.dp), // 안쪽 여백
                    contentAlignment = Alignment.Center // 텍스트 가운데 정렬
                ) {
                    Text(
                        text = text,
                        color = WH,
                        fontFamily = PretendardFontFamily,
                        fontWeight = FontWeight.SemiBold,
                        fontSize = 12.sp,
                        maxLines = 1
                    )
                }
}
```

형태로 Reusable한 컴포저블로 만들어서 사용하세요. 아래 보니깐 중복되네요~!

DJDrama replied:
추가적으로 color=WH에서 WH같이 직관적이지 않은 이름보다 직관적인 이름을 사용하는 것으로 해보세요~! 제가 느끼기엔 WH=WHITE 느낌인데, 차라리 White으로 한다던지요~!

agvber replied:
> 추가적으로 color=WH에서 WH같이 직관적이지 않은 이름보다 직관적인 이름을 사용하는 것으로 해보세요~! 제가 느끼기엔 WH=WHITE 느낌인데, 차라리 White으로 한다던지요~!

color 네이밍은 피그마에 있는 color 이름을 그대로 사용하다 보니 그렇게 된 것 같아요!
피그마로 디자인 작업하다보면 color 이름이 같으면 바로바로 매치되기 쉽다고 생각해서 지정했습니다!

그래도 변경하는 것이 좋을까요??


![image](https://github.com/user-attachments/assets/0665aeea-55bb-471d-99e1-fb866dd6e284)

DJDrama replied:
이건 마음대로 하시면 됩니다~! 보여서 코멘트해보았어요 ㅋㅋ

DJDrama replied:
@agvber

DJDrama commented on core/designsystem/src/main/java/com/teampatch/core/designsystem/component/TempMemoryCard.kt:
```diff
@@ -0,0 +1,94 @@
+        )
+        Column(
+            modifier = Modifier.padding(horizontal = 14.dp, vertical = 12.dp)
+        ) {
```
불필요한 Column 같아요~!
하위 Text에 패딩을 주던지 
Spacer을 사용하는 것도 고려해보세요~!

Text의 Modifier을 padding(horizontal=14.dp)를 주시고
Spacer(modifier = Modifier.height(height=12.dp)를 주시구

DJDrama commented on feature/memorystorage-detail/src/main/java/com/teampatch/memorystorage/feature/detail/MemoryStorageDetailUiState.kt:
```diff
@@ -0,0 +1,26 @@
+    Conversation,
+}
+
+sealed class MemoryStorageDetailUiState(
```
굳이 subclass에 default 값을 줄거면 sealed interface로 해도 상관없지 않을까요? 물론 전체 코드를 안봐서 정확히는 모르지만..

DJDrama commented on feature/memorystorage/src/main/java/com/teampatch/feature/memorystorage/MemoryStorageUiState.kt:
```diff
@@ -0,0 +1,9 @@
+
+import com.teampatch.core.domain.model.MemoryCard
+
+sealed class MemoryStorageUiState {
```
sealed interface

agvber commented:
수고하셨습니다😄😄
수정되면 재요청해주세요!!

theBettor committed:
fix: navigation

theBettor committed:
refactor: Reusable한 컴포저블로 만들어서 사용하기

theBettor committed:
fix: 불필요한 Column 삭제

theBettor committed:
fix: 상태표현만 하니까 sealed interface로

로직이 없는 상태 표현만 있다면 → sealed interface로 가볍게.

공통 로직이나 동작을 포함해야 한다면 → sealed class 유지.

theBettor committed:
fix : apply로 변경

apply: 객체 속성 설정용 (this 리시버)

also: side effect용 (it 리시버)

agvber reviewed (approved):
우왕 수고 하셨습니다!!!
